cmd/skupper/skupper_cluster_test.go

```diff
@@ -0,0 +1,1579 @@
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					"application": "tcp-go-echo-ss",
+				},
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:            "tcp-go-echo",
+						Image:           "quay.io/skupper/tcp-go-echo",
+						ImagePullPolicy: corev1.PullIfNotPresent,
+						Ports: []corev1.ContainerPort{
+							{
+								Name:          "tcp-go-echo",
+								Protocol:      corev1.ProtocolTCP,
+								ContainerPort: 9090,
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+}
+
+func TestInitInteriorWithCluster(t *testing.T) {
+	testcases := []struct {
+		doc             string
+		args            []string
+		expectedCapture string
+		expectedOutput  string
+		expectedError   string
+		realCluster     bool
+	}{
+		{
+			doc:             "init-test1",
+			args:            []string{"--help"},
+			expectedCapture: "",
+			expectedOutput:  "Setup a router and other supporting objects to provide a functional skupper",
+			expectedError:   "",
+			realCluster:     false,
+		},
+	}
+
+	namespace = "init-interior-cluster-test-" + strings.ToLower(utils.RandomId(4))
+	kubeContext = ""
+	kubeConfigPath = ""
+
+	if *clusterRun {
+		cli = NewClient(namespace, kubeContext, kubeConfigPath)
+	} else {
+		cli = newMockClient(namespace)
+	}
+
+	if c, ok := cli.(*client.VanClient); ok {
+		_, err := kube.NewNamespace(namespace, c.KubeClient)
+		assert.Check(t, err)
+		defer kube.DeleteNamespace(namespace, c.KubeClient)
+	}
+
+	for _, tc := range testcases {
+		if tc.realCluster && !*clusterRun {
+			continue
+		}
+
+		cmd := NewCmdInit(testClient)
+		silenceCobra(cmd)
+		testCommand(t, cmd, tc.doc, tc.expectedError, tc.expectedCapture, tc.expectedOutput, tc.args...)
+	}
+}
+
+func TestDeleteWithCluster(t *testing.T) {
+	testcases := []struct {
+		doc             string
+		args            []string
+		expectedCapture string
+		expectedOutput  string
+		expectedError   string
+		realCluster     bool
+	}{
+		{
+			doc:             "delete-test1",
+			args:            []string{"--help"},
+			expectedCapture: "",
+			expectedOutput:  "delete will delete any skupper related objects from the namespace",
+			expectedError:   "",
+			realCluster:     false,
+		},
+		{
+			doc:             "delete-test2",
+			args:            []string{},
+			expectedCapture: "Skupper is now removed from",
+			expectedOutput:  "",
+			expectedError:   "",
+			realCluster:     true,
+		},
+	}
+
+	namespace = "cmd-delete-cluster-test-" + strings.ToLower(utils.RandomId(4))
+	kubeContext = ""
+	kubeConfigPath = ""
+
+	if *clusterRun {
+		cli = NewClient(namespace, kubeContext, kubeConfigPath)
+	} else {
+		cli = newMockClient(namespace)
+	}
+
+	if c, ok := cli.(*client.VanClient); ok {
+		_, err := kube.NewNamespace(namespace, c.KubeClient)
+		assert.Check(t, err)
+		defer kube.DeleteNamespace(namespace, c.KubeClient)
+	}
+	skupperInit(t, []string{"--edge"}...)
```

Comment:
if I understand correctly, the "delete" table case only works once, because skupper init is executed only once

Reply:
which for now, does not produce any error

Reply:
That is correct. If needed for coverage, init could be moved to the test loop in a future patch.